api/envoy/api/v2/cds.proto

```diff
@@ -777,6 +777,9 @@ message Cluster {
   // maybe by allowing LRS to go on the ADS stream, or maybe by moving some of the negotiation
   // from the LRS stream here.]
   core.ConfigSource lrs_server = 42;
+
+  //Always use TCP queries instead of UDP queries for DNS lookups.
+  bool use_tcp_for_dns_lookups = 43;
```

Comment:
Can you locate this next to the other DNS options (e.g. TTL) above? Also, needs a `[#not-implemented-hide:]` comment for now, and a `[#next-major-version: reconcile DNS options in a single message]` comment. Thanks.
/wait